test/stability/testMerge/testMerge.go

```diff
@@ -89,151 +123,440 @@ func main() {
 	target := flag.String("target", "", "Target host name from config file (mandatory for VM)")
 	flag.StringVar(&dpdkLogLevel, "dpdk", "--log-level=0", "Passes an arbitrary argument to dpdk EAL")
 	flag.Parse()
-	if err := executeTest(*configFile, *target, testScenario); err != nil {
-		fmt.Printf("fail: %+v\n", err)
+	flow.CheckFatal(initDPDK())
+	if useSimpleGenerate {
+		generatorType = generate
+	} else {
+		generatorType = fastGenerate
 	}
+	flow.CheckFatal(executeTest(*configFile, *target, testScenario, pipelineScenario(pipelineType), generatorType, addSegmAfterMerge))
 }
 
-func executeTest(configFile, target string, testScenario uint) error {
-	if testScenario > 3 || testScenario < 0 {
-		return errors.New("testScenario should be in interval [0, 3]")
-	}
-	// Init NFF-GO system
-	config := flow.Config{
-		DPDKArgs: []string{dpdkLogLevel},
+// getter - is a part of pipeline getting packets by receive or generate or fast generate
+type getScenario uint
+
+const (
+	generate getScenario = iota
+	fastGenerate
+	recv
+)
+
+func setGetter(scenario getScenario, inputSource uint) (finalFlow *flow.Flow, err error) {
+	switch scenario {
+	case recv:
+		finalFlow, err = flow.SetReceiver(uint16(inputSource))
+	case fastGenerate:
+		if inputSource == useGenerator1 {
+			finalFlow, err = flow.SetFastGenerator(generatePacketGroup, speed, &generatorParameters{isGroup1: true})
+		} else if inputSource == useGenerator2 {
+			finalFlow, err = flow.SetFastGenerator(generatePacketGroup, speed, &generatorParameters{isGroup1: false})
+		} else {
+			return nil, errors.New(" setGetter: unknown generator type")
+		}
+	default:
+		if inputSource == useGenerator1 {
+			finalFlow = flow.SetGenerator(generatePacketGroup, &generatorParameters{isGroup1: true})
+		} else if inputSource == useGenerator2 {
+			finalFlow = flow.SetGenerator(generatePacketGroup, &generatorParameters{isGroup1: false})
+		} else {
+			return nil, errors.New(" setGetter: unknown generator type")
+		}
 	}
-	if err := flow.SystemInit(&config); err != nil {
-		return err
+	return finalFlow, err
+}
+
+// segment type definition, not all possible values are used to decrease complexity
+type segmentScenario uint
+
+const (
+	handle segmentScenario = iota
+	handleDrop
+	split
+)
+
+// returns nil if no new flow was created and new flow if was
+func setSegment(scenario segmentScenario, f *flow.Flow) (finalFlow *flow.Flow, err error) {
```

Comment:
You change one function setSegment to one final function Set something. It is confusing. Please, remove setSegment function and set functions directly in their places. Also, if you post that you place "split" you should honestly test both flows. If you don't want to have split logic - place partition instead.